addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -226,19 +218,27 @@ public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, St
 
     @Override
     public FritzahaWebInterface getWebInterface() {
-        return this.connection;
+        if (getThing().getThingTypeUID().equals(PL546E_STANDALONE_THING_TYPE)) {
+            return connection;
+        } else {
+            Bridge bridge = getBridge();
+            if (bridge != null) {
+                BridgeHandler handler = bridge.getHandler();
+                if (handler != null && handler instanceof BoxHandler) {
+                    return ((BoxHandler) handler).getWebInterface();
+                }
+            }
+        }
+        return null;
```

Comment:
You could consider throwing some really specific exception in this case and catch that somewhere in the end of the calling method. Because you return null now you have to handle it immediately, catching the errors can be more easily separated from the program flow.